airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConfig.java

```diff
@@ -126,15 +133,24 @@ public S3FormatConfig getFormatConfig() {
     return formatConfig;
   }
 
-  public AmazonS3 getS3Client() {
```

Comment:
Currently we are creating a new S3 client for each stream, which is totally unnecessary. S3 client is thread safe, and it comes with a connection pool. So we should always reuse one client.